package.json

```diff
@@ -15,7 +15,8 @@
     "test": "gulp test lint",
     "test:smoke": "gulp test",
     "test:integration": "gulp test:integration",
-    "format": "find src gulpfile.js \\( -iname '*.ts' -o -iname '*.js' \\) -not -path '*fixtures*' | xargs clang-format --style=file -i"
+    "format": "find src gulpfile.js \\( -iname '*.ts' -o -iname '*.js' \\) -not -path '*fixtures*' | xargs clang-format --style=file -i",
+    "test:watch": "tsc-then -- mocha -c --ui tdd lib/test/integration/*_test.js lib/test/unit/*_test.js lib/test/unit/*/*_test.js"
```

Comment:
You could shorten this line by using a more complex glob selector if you want,

```
"test:watch": "tsc-then -- mocha -c --ui tdd 'lib/test/@\\(integration\\|unit\\)/{,!\\(fixtures\\)/**/}/*_test.js'"
```

Reply:
The gulp command does too much, it compiles which we're using tsc-then to do quickly and incrementally. 

Gulp also doesn't seem to do deduplication, so since both `test` and `test:integration` both depend on `build` it will end up compiling _twice_.

I often feel like I'm holding gulp wrong though, there's got to be a way to use it so that it doesn't do this… I would think.

Reply:
I definitely thought `gulp test test:integration` would only run build once, maybe I'm wrong